### ISSUE TITLE
How to restricted API Key for android and iOS

### ISSUE THREAD

**Issue by ajaysaini-sgvu** [FaridSafi/react-native-google-places-autocomplete#489]
Hello Team,

I am using this plugin for places suggestions. I have restricted by API Key using SHA1 and bundle id on google developer console.

I am getting below error after setting restrictions:

>  WARN  google places autocomplete: This IP, site or mobile application is not authorized to use this API key. Request received from IP address X.X.X.X, with referer: undefined

I tried by setting the origin like, then it doesn't give list of places.

```
query={{
	key: this.googleApiKey,
	components: `country:${componentRestrictions.country[0]}`,
	types: 'address',
	origin: 'https://www.xxxx.com'
}}
```
curl -H "referer:https://www.xxxx.com" https://maps.googleapis.com/maps/api/place/autocomplete/json?&input=123&key=xxxxxxxxxx&components=country%3AUS&types=address&origin=https%3A%2F%2Fwww.xxxx.com

Returns this error

```
error_message:Invalid request. Invalid 'origin' parameter.
predictions:
status:INVALID_REQUEST
```

Can you please suggest if it possible to restrict it ?

**Comment by jparksecurity** [comment]
I don't think the lib supports restricted API key.

According to [the Google doc](https://developers.google.com/places/web-service/autocomplete), `origin` parameter is 

>origin — The origin point from which to calculate straight-line distance to the destination (returned as distance_meters). If this value is omitted, straight-line distance will not be returned. Must be specified as latitude,longitude.

Hence, it has to be

curl -H "referer:https://www.xxxx.com" https://maps.googleapis.com/maps/api/place/autocomplete/json?&input=123&key=xxxxxxxxxx&components=country%3AUS&types=address

which is not how the lib curruntly works

**Comment by bell-steven** [comment]
Closed by #526
